x/ccv/consumer/keeper/relay.go

```diff
@@ -223,6 +236,40 @@ func (k Keeper) SendPackets(ctx sdk.Context) {
 // in conjunction with the ibc module's execution of "acknowledgePacket",
 // according to https://github.com/cosmos/ibc/tree/main/spec/core/ics-004-channel-and-packet-semantics#processing-acknowledgements
 func (k Keeper) OnAcknowledgementPacket(ctx sdk.Context, packet channeltypes.Packet, ack channeltypes.Acknowledgement) error {
+	if res := ack.GetResult(); res != nil {
+		if len(res) != 1 {
+			k.Logger(ctx).Error("recv invalid ack; expected length 1", "channel", packet.SourceChannel, "ack", res)
```

Comment:
Shouldn't we also return an error here?

Reply:
Is potentially the idea here that we do not return errors because the code that follows does `if err := ack.GetError(); err != "" {`?
Does this mean that we could have `ack.GetResult() != nil` being true and also `ack.GetError() != nil`? If this was the case, I guess then we shouldn't return `nil` later on in:
```
		if consumerPacket.Type == ccv.VscMaturedPacket {
			return nil
		}
```
because this would prevent checking for `ack.GetError()`?

Additionally, in the case where `ack.GetError() != nil` we close the underlying channel. Is this something we would like to do here as well?

Reply:
By checking ibc-go source code, it's impossible for `ack.GetResult()` and `ack.GetError()` to both return non-nil values for the same `ack`. 

Additionally, returning an error from this method [seems to just log again from ibc-go](https://github.com/cosmos/ibc-go/blob/951b3abd9f38af0697a6d65b95fd2b0921547acd/modules/core/keeper/msg_server.go#L680). I'm fine returning the error and logging again, just seems redundant. 

Re closing the underlying channel, [this issue is relevant ](https://github.com/cosmos/interchain-security/issues/547)

Reply:
> Additionally, returning an error from this method [seems to just log again from ibc-go](https://github.com/cosmos/ibc-go/blob/951b3abd9f38af0697a6d65b95fd2b0921547acd/modules/core/keeper/msg_server.go#L680). I'm fine returning the error and logging again, just seems redundant.

If we don't return an error, then IBC will log `acknowledgement succeeded`, which will make debugging very annoying.

Reply:
Ah valid point! Fixed in https://github.com/cosmos/interchain-security/commit/afbe04c03a6ba00b43b7f9bed6d658e307a2dac0

Reply:
Unfortunately the above change caused an odd bug that only fails integration tests in CI, and not locally. I'm looking into it